pkg/infradb/manager.go

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2023 Nordix Foundation.
+
+package infradb
+
+import (
+	"encoding/json"
+	"log"
+
+	badger "github.com/dgraph-io/badger/v4"
+	pb "github.com/opiproject/opi-api/network/evpn-gw/v1alpha1/gen/go"
+)
+
+var db *badger.DB
+
+func Init() {
+
+	var err error
+	// Open the Badger database located in the /tmp/badger directory.
+	// It will be created if it doesn't exist.
+	db, err = badger.Open(badger.DefaultOptions("/tmp/badger"))
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+/*func Close() {
+	// Close the DB
+	db.Close()
+}*/
+
+// TransactionManager provides a wrapper for DB transactions
+type TransactionManager interface {
+	Add() error
+	Update() error
+	Read(resourceName string) error
+	Delete() error
+}
+
+type VrfManager struct {
+	VrfObj *pb.Vrf
+}
+
+func (v *VrfManager) Add() error {
+
+	err := db.Update(func(txn *badger.Txn) error {
+		value, err := json.Marshal(v.VrfObj)
```

Comment:
according to https://pkg.go.dev/google.golang.org/protobuf/encoding/protojson
```
This package produces a different output than the standard "encoding/json" package, which does not operate correctly on protocol buffer messages.
```
so please use
```
import "google.golang.org/protobuf/proto"
value, err := proto.Marshal(v.VrfObj)
```
instead

Reply:
I will fix that